packages/compiler/src/server/serverlib.ts

```diff
@@ -525,7 +525,10 @@ export function createServer(host: ServerHost): Server {
             diagnostic.severity = DiagnosticSeverity.Hint;
           }
         }
-        diagnostic.data = { id: diagnosticIdCounter++ };
+        diagnostic.data = {
+          id: diagnosticIdCounter++,
+          file: await fileService.getPath({ uri: diagDocument.uri }),
```

Comment:
We may not need to set `file` for each diagnostic. The `file` is only useful for the import-not-found diagnostic, right?